File: Neanias.Accounting.Service.WebApp/src/app/core/query/service-reset-entry-sync.lookup.ts
import { IsActive } from '@app/core/enum/is-active.enum';
import { Lookup } from '@common/model/lookup';
import { Guid } from '@common/types/guid';

export class ServiceResetEntrySyncLookup extends Lookup implements ServiceResetEntrySyncFilter {
	ids: Guid[];
	excludedIds: Guid[];
	like: string;
	isActive: IsActive[];

	constructor() {
		super();
	}
}

export interface ServiceResetEntrySyncFilter {
	ids: Guid[];
	excludedIds: Guid[];
	like: string;
	isActive: IsActive[];
}

File: Cite.Accounting.Service/Query/ServiceResetEntrySyncLookup.cs
using Cite.Accounting.Service.Common;
using Cite.Tools.Data.Query;
using System;
using System.Collections.Generic;

namespace Cite.Accounting.Service.Query
{
	public class ServiceResetEntrySyncLookup : Lookup
	{
		public List<Guid> Ids { get; set; }
		public String Like { get; set; }
		public List<IsActive> IsActive { get; set; }

		public ServiceResetEntrySyncQuery Enrich(QueryFactory factory)
		{
			ServiceResetEntrySyncQuery query = factory.Query<ServiceResetEntrySyncQuery>();

			if (this.Ids != null) query.Ids(this.Ids);
			if (this.IsActive != null) query.IsActive(this.IsActive);
			if (!String.IsNullOrEmpty(this.Like)) query.Like(this.Like);

			this.EnrichCommon(query);

			return query;
		}
	}
}
